public/js/controllers/attackcost_controller.js

```diff
@@ -236,6 +248,20 @@ export default class extends Controller {
     this.setActivePoint()
   }
 
+  updateQueryString () {
+    let updatedSettings = {}
+    const settingsEntries = Object.entries(this.settings)
+    const defaultSettingsEntries = Object.entries(this.defaultSettings)
+    for (const [key, val] of settingsEntries) {
+      let defaultValue = _.find(defaultSettingsEntries, (entry) => {
+        return entry[0] === key
+      })
+      if (val === null || val.toString() === defaultValue[1].toString()) continue
+      updatedSettings[key] = val
+    }
+    this.query.replace(updatedSettings)
```

Comment:
How about

```
const [query, settings, defaults] = [{}, this.settings, this.defaultSettings]
for (const k in settings) {
  if (!settings[k] || settings[k] === defaults[k]) continue
  query[k] = settings[k]
}
this.query.replace(query)
```

You may need a string conversion for comparison, but ideally, the `this.settings` would be assigned with the correct type, e.g. using `parseInt` anyway.